src/main/scala/com/databricks/spark/xml/parsers/StaxXmlParser.scala

```diff
@@ -44,14 +44,15 @@ private[xml] object StaxXmlParser extends Serializable {
       xml: RDD[String],
       schema: StructType,
       options: XmlOptions): RDD[Row] = {
-    def failedRecord(record: String): Option[Row] = {
+    def failedRecord(record: String, cause: Throwable = null): Option[Row] = {
       // create a row even if no corrupt record column is present
       options.parseMode match {
         case FailFastMode =>
           throw new RuntimeException(
-            s"Malformed line in FAILFAST mode: ${record.replaceAll("\n", "")}")
+            s"Malformed line in FAILFAST mode: ${record.replaceAll("\n", "")}", cause)
         case DropMalformedMode =>
-          logger.warn(s"Dropping malformed line: ${record.replaceAll("\n", "")}")
+          val reason = if (cause != null) cause.getMessage else "Unknown";
```

Comment:
Tiny nit: You could print either "Reason: ${cause.getMessage}" or nothing at all instead of printing "Reason: Unknown" in the latter case